pagopa/io-app
# chore: [IOBP-625] Add guest payment flow to the new wallet section

## Short description
This PR introduces the guest payment flow, allowing an user to pay with a payment method not saved in its wallet.

## List of changes proposed in this pull request
- Refactoring/reorganization of payment checkout actions
- Moved PSP sorting/selection logic from sagas to reducer
- Add guest payment method fetch in payment methods screen
- Removed "missing payment methods" error screen

## How to test
You can test this PR on both **local** and **UAT** environments:
- Local
Checkout this PR on io-dev-api-server: https://github.com/pagopa/io-dev-api-server/pull/363/
Start a payment flow and select a payment method not saved to your wallet.
- UAT
Start a payment flow using this data:
  - **Payment number**: 30200000##########
  - **Fiscal code**: 77777777777

You should be able to finalize a payment.

## Preview

https://github.com/pagopa/io-app/assets/6160324/2329ef6c-9607-4445-985f-a7592b56a66d
